agent/consul/kvs_endpoint.go

```diff
@@ -160,18 +160,13 @@ func (k *KVS) Get(args *structs.KeyRequest, reply *structs.IndexedDirEntries) er
 			}
 
 			if ent == nil {
-				// Must provide non-zero index to prevent blocking
-				// Index 1 is impossible anyways (due to Raft internals)
-				if index == 0 {
-					reply.Index = 1
-				} else {
-					reply.Index = index
-				}
+				reply.Index = index
```

Comment:
Presumably this part of the change was intentional?

Reply:
Ya, this is now handled generally in `Server.blockingQuery` by calling `Server.setQueryMeta`, and setting the value to 1 if it is still 0.  I guess this logic in `kvs_endpoint` predated the general fix in `blockingQuery`.